build fails with aeson-2.0

```
spacecookie                    > /tmp/stack-e91b7c5fd4ba17cb/spacecookie-1.0.0.0/server/Network/Spacecookie/Config.hs:28:28: error:
spacecookie                    >     • Couldn't match expected type ‘Key’ with actual type ‘Text’
spacecookie                    >     • In the second argument of ‘(.:?)’, namely ‘x’
spacecookie                    >       In the expression: v .:? x
spacecookie                    >       In an equation for ‘maybePath’: maybePath [x] v = v .:? x
spacecookie                    >    |
spacecookie                    > 28 | maybePath [x]    v = v .:? x
spacecookie                    >    |                            ^
spacecookie                    > 
spacecookie                    > /tmp/stack-e91b7c5fd4ba17cb/spacecookie-1.0.0.0/server/Network/Spacecookie/Config.hs:29:28: error:
spacecookie                    >     • Couldn't match expected type ‘Key’ with actual type ‘Text’
spacecookie                    >     • In the second argument of ‘(.:?)’, namely ‘x’
spacecookie                    >       In the first argument of ‘(>>=)’, namely ‘v .:? x’
spacecookie                    >       In the expression: v .:? x >>= fmap join . traverse (maybePath xs)
spacecookie                    >    |
spacecookie                    > 29 | maybePath (x:xs) v = v .:? x >>= fmap join . traverse (maybePath xs)
spacecookie                    >    |                            ^
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,11 @@
 # Revision history for spacecookie
 
+## 1.0.0.1
+
+2021-11-29
+
+This release fixes compilation with `aeson >= 2.0`.
+
 ## 1.0.0.0
 
 2021-03-16

--- a/server/Network/Spacecookie/Config.hs
+++ b/server/Network/Spacecookie/Config.hs
@@ -1,4 +1,5 @@
 {-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE CPP               #-}
 module Network.Spacecookie.Config
   ( Config (..)
   , LogConfig (..)
@@ -23,7 +24,13 @@ data Config
   , logConfig     :: LogConfig
   }
 
+-- We only use string literals with 'maybePath', so we can just switch between
+-- Key and Text, since both have an IsString instance for OverloadedStrings.
+#if MIN_VERSION_aeson(2,0,0)
+maybePath :: FromJSON a => [Key] -> Object -> Parser (Maybe a)
+#else
 maybePath :: FromJSON a => [Text] -> Object -> Parser (Maybe a)
+#endif
 maybePath []     _ = fail "got empty path"
 maybePath [x]    v = v .:? x
 maybePath (x:xs) v = v .:? x >>= fmap join . traverse (maybePath xs)

--- a/spacecookie.cabal
+++ b/spacecookie.cabal
@@ -1,6 +1,6 @@
 cabal-version:       3.0
 name:                spacecookie
-version:             1.0.0.0
+version:             1.0.0.1
 synopsis:            Gopher server library and daemon
 description:         Simple gopher library that allows writing custom gopher
                      applications. Also includes a fully-featured gopher server
@@ -52,7 +52,7 @@ executable spacecookie
   import:              common-settings, gopher-dependencies
   main-is:             Main.hs
   build-depends:       spacecookie
-                     , aeson >= 1.5 && < 2.0
+                     , aeson >= 1.5
                      , systemd >= 2.1.0
                      , fast-logger >= 2.4.0
   hs-source-dirs:      server

